### ISSUE TITLE
Enrolment date in MP should be taken from Redcap

### ISSUE THREAD

**Issue by yatharthranjan** [RADAR-CNS/RADAR-RedcapIntegration#16]
The date for subject enrolment should not be generated within the Management Portal but instead should be taken from Redcap which already has this date. Right now there are 3 dates in the system for a single enrolment event(2 in redcap and 1 in MP). This will probably require a GET request using the Redcap API since Redcap data trigger does not supply this information.

**Comment by afolarin** [comment]
Yes, this is generally a weak point for the REDCap <--> Management Portal integration:
There are 3 different dates:
1) REDCap Enrollment (manually entered)
2) REDCap Calendar Scheduling date (purely administrative for the calculated schedule based on events, this has nothing to do with the integration but I list it here as it is again separate from and possibly inconsistent with (1)
3) The Management Portal record creation date

The semantics of these dates is slightly problematic as they can under some situations be inconsistent. particularly 1 & 3) being most problematic.

3) is used by apps as the enrollment date
1) is used by clinical teams and data analysts as the enrollment date

example scenario:
c4d22114...MDD OSR Dis
we can see that the record creation in MP was 24/08/18 this is after the enrolment date 22/06/18  it's probably a typo in the enrollment date Or the REDCap record was created and the integration initially failed to create the corresponding Management Portal record, afterwards then someone re-attempted this successfully 63days late

Almost certain this could be improved, 
1) warn if they are inconsistent
2) keep different dates with different semantics (Redcap record creation, Redcap enrollment, Management Portal Record Creation, etc...) i.e. so they are noted not to be "enrollment", which may be misleading (e.g. a record could be created then an enrollment take place at a future time)
2) try to provide a single source for Enrollment date in both Redcap and Management Portal *this is probably desirable but harder (perhaps Management Portal could pull the Enrollment Date via the REDCAp API if we embed or copy this into the RADAR Integration form?

**Comment by yatharthranjan** [comment]
Just a techincal constraint. The subject `createDate` in Management portal is automatically created by the auditing system and will not be available for update using REST endpoints.
So perhaps the opposite workflow would be better, i.e. To update the Redcap Enrolment Date field after creation success on Management portal using the MP subject's createDate.

**Comment by callumstew** [comment]
Two things to consider are:
- changing MP date based on Redcap will change the participant's active app schedule whenever it next needs to checks the date
- changing Redcap data based on MP could work in general, but many participant's have an incorrect createDate because they were all created when MP was migrated in Aug 2018, after their actual enrolments, so you would lose their actual enrolment date
